packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts

```diff
@@ -1,30 +1,66 @@
 import { GenerationLanguage } from "@fern-api/generators-configuration";
 import { Name, NameAndWireValue, SafeAndUnsafeString } from "@fern-fern/ir-sdk/api";
-import { camelCase, snakeCase, upperFirst } from "lodash-es";
+import { camelCase, snakeCase, upperFirst, words } from "lodash-es";
 import { RESERVED_KEYWORDS } from "./reserved";
 
 export interface CasingsGenerator {
     generateName(name: string): Name;
     generateNameAndWireValue(args: { name: string; wireValue: string }): NameAndWireValue;
 }
 
-export function constructCasingsGenerator(generationLanguage: GenerationLanguage | undefined): CasingsGenerator {
+export function constructCasingsGenerator(
+    generationLanguage: GenerationLanguage | undefined,
+    specialCasing: boolean
+): CasingsGenerator {
```

Comment:
nit: i think using named params here will be better 

```
{ generationLanguage, specialCasing }: { generationLanguage: GenerationLanguage | undefined | specialCasing: boolean } 
```

Reply:
Good call, always a better move to do this when we can.